pydanclick/model/model_conversion.py

```diff
@@ -76,5 +78,11 @@ def func(**kwargs):
         shorten=cast(Optional[Dict[DottedFieldName, OptionName]], shorten),
         extra_options=cast(Dict[DottedFieldName, _ParameterKwargs], extra_options),
     )
-    validator = functools.partial(model_validate_kwargs, model=model, qualified_names=qualified_names)
+
+    defaults_to_ignore = {}
+    if defer_set_default:
```

Comment:
I was thinking of maybe changing this to a different implementation where the defaults are not even added to click instead of comparing to the default value in validation.py.

Conceptually I think that it makes more sense because a CLI default may not communicate the correct behavior in this case.

What are your thoughts?

Reply:
Good point, would be more robust indeed. We could have a sentinel value to detect unset default in those cases.

```python
unset = object()
```

then simply check `value is unset` to detect unset values.

Reply:
I ended up rewriting the implementation a bit to better manage the defaults behavior.

In a nutshell, I made it so that the type is wrapped in PydanclickParamType. This allows overriding the behavior of the convert function when a PydanclickDefault type is passed. It will return None instead of the actual value on the convert method when a PydanclickDefault is being converted.

This has the effect of delegating to the pydantic model.

The other aspect which is better handled is when "show_default" is passed into click. The PydanclickDefault will actually show the default because it overrides the __repr__ function and proxies the actual version for the default value.